pkg/operator/ceph/cluster/cluster.go

```diff
@@ -66,13 +65,14 @@ func newCluster(c *cephv1.CephCluster, context *clusterd.Context) *cluster {
 		// at this phase of the cluster creation process, the identity components of the cluster are
 		// not yet established. we reserve this struct which is filled in as soon as the cluster's
 		// identity can be established.
-		Info:      nil,
-		Namespace: c.Namespace,
-		Spec:      &c.Spec,
-		context:   context,
-		stopCh:    make(chan struct{}),
-		ownerRef:  ownerRef,
-		mons:      mon.New(context, c.Namespace, c.Spec.DataDirHostPath, c.Spec.Network.HostNetwork, ownerRef),
+		Info:                 nil,
+		Namespace:            c.Namespace,
+		Spec:                 &c.Spec,
+		context:              context,
+		stopCh:               make(chan struct{}),
+		ownerRef:             ownerRef,
+		orchestrationPending: true,
```

Comment:
Could we get a comment here to explain that this is initialized to true so that the first call to orchestrate the cluster will go through?

Reply:
I think it's unnecessary to initialize this to true anyway. Any call to `createInstance` (or `handleUpdate` automatically implies that at least one update is pending, the one that spawned the call to the method. Upon entry to the method, I would set (atomically) orchestration pending. (Atomically) If there is already an orchestration running, the function can return without error. (Atomically) If there is not an orchestration running and there is an orchestration pending, then set running true and pending false. Deep copy the spec. Once the cluster is updated, (atomically) set orchestration running false. The loop will check again (atomically) if there is an orchestration pending and if there is not an orchestration running (another thread may have taken over the pending update).

Reply:
```go

func updateWork() retvals {
    // work
}

func updateStuff(...) {
    pending = true
    for atomic(pending==true && running==false, running=true, pending=false) {
        // deep copy spec
        retvals = func() {
            defer atomic(running=false)
            return updateWork()
        }
    }
}
```

Reply:
Setting `c.orchestrationPending` here make sense. I will remove the initialization.